tests/unit/test_remote_configuration.py

```diff
@@ -0,0 +1,323 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import unittest
+from unittest.mock import Mock, PropertyMock, call, patch
+
+import yaml
+from charms.alertmanager_k8s.v0.alertmanager_remote_configuration import (
+    DEFAULT_RELATION_NAME as RELATION_NAME,
+)
+from charms.alertmanager_k8s.v0.alertmanager_remote_configuration import (
+    AlertmanagerRemoteConfigurationConsumer,
+)
+from deepdiff import DeepDiff  # type: ignore[import]
+from helpers import k8s_resource_multipatch
+from ops import testing
+from ops.charm import CharmBase
+from ops.model import BlockedStatus
+
+from charm import AlertmanagerCharm
+
+testing.SIMULATE_CAN_CONNECT = True
+
+TEST_APP_NAME = "consumer-tester"
+METADATA = f"""
+name: {TEST_APP_NAME}
+requires:
+  {RELATION_NAME}:
+    interface: alertmanager_remote_configuration
+    limit: 1
+"""
+TEST_ALERTMANAGER_CONFIG_FILE = "/test/rules/dir/config_file.yml"
+TEST_ALERTMANAGER_DEFAULT_CONFIG = """route:
+  receiver: dummy
+receivers:
+- name: dummy
+"""
+TEST_ALERTMANAGER_REMOTE_CONFIG = """receivers:
+- name: test_receiver
+route:
+  receiver: test_receiver
+  group_by:
+  - alertname
+  group_wait: 1234s
+  group_interval: 4321s
+  repeat_interval: 1111h
+"""
+
+
+class RemoteConfigurationConsumerCharm(CharmBase):
+    ALERTMANAGER_CONFIG_FILE = "./tests/unit/test_config/alertmanager.yml"
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.remote_configuration_consumer = AlertmanagerRemoteConfigurationConsumer(
+            charm=self,
+            relation_name=RELATION_NAME,
+            config_file_path=self.ALERTMANAGER_CONFIG_FILE,
+        )
+
+
+class TestAlertmanagerRemoteConfigurationConsumer(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = testing.Harness(RemoteConfigurationConsumerCharm, meta=METADATA)
+        self.addCleanup(self.harness.cleanup)
+        testing.SIMULATE_CAN_CONNECT = True
+        self.harness.set_leader(True)
+        self.harness.begin()
+
+    def test_given_remote_configuration_consumer_charm_providing_config_without_templates_when_relation_joined_then_alertmanager_config_is_updated_in_the_relation_data_bag(  # noqa: E501
```

Comment:
Method name is too long.
Please move the gherkin statements into individual comments.

Reply:
I disagree, we're not in any shortage of characters to use :D

Reply:
I'll let you guys find a common ground here ;)
In our team we don't care about the length of the test method name as long as it correctly describes what the test does.

Reply:
`test_config_without_templates_updates_databag` would be a reasonable alternative name :)

Reply:
All names updated :)